resolevd-ts: add resolved-ts related config doc (#5792)
pingcap/docs: tikv-configuration-file.md

+ Default value: `6`, which means 6 tasks can be concurrent executed at most.
+ Note: The value of `incremental-scan-concurrency` must be greater than or equal to that of `incremental-scan-threads`; otherwise, TiKV will report an error at startup.

## resolved-ts

Configuration items related to maintaining the Resolved TS to serve Stale Read requests.

### `enable`

> + Enables whether to maintain the Resolved TS for all Regions.

Reviewer: ```suggestion
+ Determines whether to maintain the Resolved TS for all Regions.
```

> + Default value: `false`

Reviewer: ```suggestion
+ Default value: `true`
```

### `advanced-ts-interval`

> + The time interval for regularly advancing the Resolved TS.

Reviewer: ```suggestion
+ The time interval at which the Resolved TS is processed.
```

Reviewer 2: Perhaps we can reuse the description of `cdc.min-ts-interval`, they are used for the same goal.

+ Default value: `"1s"`

### `scan-lock-pool-size`

> + The number of threads that TiKV scans for MVCC (multi-version concurrency control) lock data when initializing the Resolved TS.

Reviewer: @NingLin-P PTAL whether this comment is correct or not. Thanks!
```suggestion
+ The number of threads that TiKV uses to scan the data locked by MVCC (multi-version concurrency control) when initializing the Resolved TS.
```

Reviewer 2: I think `MVCC (multi-version concurrency control) lock data` is more proper, TiKV only scan the lock record like `(key, lock info)`, and won't scan the `(key, value)` data
